src/Spago/Command/Build.purs

```diff
@@ -171,11 +179,48 @@ getBuildGlobs { selected, dependencies, withTests, depsOnly } =
   Set.fromFoldable $ projectGlobs <> monorepoPkgGlobs <> dependencyGlobs <> [ BuildInfo.buildInfoPath ]
   where
   -- Here we select the right globs for a monorepo setup with a bunch of packages
-  projectGlobs = join case depsOnly of
+  projectGlobs = case depsOnly of
     true -> []
     false ->
       -- We just select all the workspace package globs, because it's (1) intuitive and (2) backwards compatible
-      map workspacePackageGlob selected
+      workspacePackageGlob selected
+
+  testGlobs = case withTests of
+    true -> WithTestGlobs
+    false -> NoTestGlobs
+
+  workspacePackageGlob :: WorkspacePackage -> Array String
+  workspacePackageGlob p = Config.sourceGlob testGlobs p.package.name (WorkspacePackage p)
+
+  { yes: monorepoPkgs, no: dependencyPkgs } = partition isWorkspacePackage
+    $ Array.mapMaybe (\pkgName -> Tuple pkgName <$> Map.lookup pkgName dependencies)
+    $ Array.fromFoldable
+    $ Map.keys
+    $ unwrap
+    $ selected.package.dependencies
+
+  -- depsOnly means "no packages from the monorepo", so we filter out the workspace packages
+  dependencyGlobs = (Tuple.uncurry $ Config.sourceGlob NoTestGlobs) =<< dependencyPkgs
+  monorepoPkgGlobs
+    | depsOnly = []
+    | otherwise = (Tuple.uncurry $ Config.sourceGlob NoTestGlobs) =<< monorepoPkgs
```

Comment:
CI is failing because of the changes I made here. The underlying issue is that the globs don't include the dependencies of the package's direct dependencies. I'm not as familiar with this part of the code, so @f-f could you give some insight as to how I should best handle this?

I basically want to write
```
Set.fromFoldable $ projectGlobs <> dependencyGlobs <> [ BuildInfo.buildInfoPath ]
```
but it seems that `dependencies` includes the dependencies for all workspace packages, not just the one for which I'm currently getting globs.

Reply:
Right, we collect all the dependencies needed in the `Spago.Command.Fetch` module - we could collect a different set of dependencies for each package there...

...however this is problematic as well: if you'd like package A to hide warnings from its dependencies, and package B to definitely show you warnings for the dependencies, then building package A first will mean that Spago will build its dependencies, hide the warnings, and keep a cache of this, so when building package B no warnings will be shown either.

A possible way out of this could be to treat the external dependencies of the project as a whole (i.e. as things are right now) - this would mean that we'd end up configuring each package PSA options separately (i.e. strict, hiding warnings, etc, but only for the code in the project), but configuring the behaviour of how to treat the warnings from dependencies at the workspace level instead.

Reply:
That sounds fine to me. Most people probably don't care about the library code anyways since it's beyond their control.